### PR TITLE
Remove JavaBeans for getter and setter method

### PR DESCRIPTION
**Pull request by sima-zhu**
When received getter and setter method, remove JavaBeans `get` and `set` if any

### CODE REVIEW

File: sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/JacksonJsonSerializer.java [hunk 1]
```diff
@@ -93,16 +95,28 @@ public String convertMemberName(Member member) {
...
         }
 
         return null;
     }
+
+    private String removePrefix(String methodName) {
+        if (methodName == null || !(methodName.contains("get") || methodName.contains("set"))) {
+            return methodName;
+        }
+        String noPrefix = methodName.replaceFirst("^get|set", "");
```
Comment (srnagar) [inline sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/JacksonJsonSerializer.java:116]: Regex should be `"^get|^set"`. Current regex will replace `isPreset` to ``isPre`.

```java
jshell> System.out.println("isPreset".replaceFirst("^get|set", ""))
isPre

```

File: sdk/core/azure-core-serializer-json-jackson/src/test/java/com/azure/core/serializer/json/jackson/JacksonPropertyNameTests.java [hunk 2]
```diff
@@ -98,10 +97,35 @@ public void testPropertyNameOnMethodName() throws NoSuchMethodException {
...
+            public String hotelName1() {
+                return hotelName;
+            }
+
+            public void hotelName2(String hotelName) {
+                this.hotelName = hotelName;
+            }
+        }
+
+        Method getterM = LocalHotel.class.getDeclaredMethod("hotelName1");
```
Comment (srnagar) [inline sdk/core/azure-core-serializer-json-jackson/src/test/java/com/azure/core/serializer/json/jackson/JacksonPropertyNameTests.java:125]: Add some tests where the property name also contains `get` and `set`.

File: sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/JacksonJsonSerializer.java (hunk 1 above)
Comment (sima-zhu) [reply sdk/core/azure-core-serializer-json-jackson/src/main/java/com/azure/core/serializer/json/jackson/JacksonJsonSerializer.java]: Use BeanUtil in jackson lib to do the check

File: sdk/core/azure-core-serializer-json-jackson/src/test/java/com/azure/core/serializer/json/jackson/JacksonPropertyNameTests.java (hunk 2 above)
Comment (sima-zhu) [reply sdk/core/azure-core-serializer-json-jackson/src/test/java/com/azure/core/serializer/json/jackson/JacksonPropertyNameTests.java:136]: More test coverage now

**Pull request merged**